### PR TITLE
langchain[patch],community[patch]: add in-memory option for unstructured loader

### LINKED ISSUE #1618: unable to import {OpenAIEmbeddings}
Im using langchain@0.0.92 with Node JS App. Checked the 'langchain' folder inside "node_modules" did not see 'embeddings' sub-folder 

### What I have tried.
- Removed langchain package, installed it again using 'npm' (didn't work)
- Removed whole 'node_modules' folder (didn't work)
- Downgraded to version 0.0.84 (didn't work either)

Any clue ?

Note: I'll post my solution if find one.

### PR DESCRIPTION
**Pull request by andrewdoro**
Fixes [#1618 ](https://github.com/langchain-ai/langchain/issues/1618)

Add support for passing buffer to unstructured loader.  

This covers cases when fetching a file externally using serverless functions. Without this change we always have to write a new file before passing the path to the loader. This is an anti pattern for serverless environments, such as Vercel.
https://github.com/orgs/vercel/discussions/241#discussioncomment-363002

Old implementation was already loading the file in memory.

Fixes # (issue)
twitter @andrewdorobantu

### CODE REVIEW

File: langchain/src/document_loaders/tests/unstructured.int.test.ts [hunk 1]
```diff
@@ -29,6 +30,34 @@ test.skip("Test Unstructured base loader", async () => {
   }
```
[the remark this answers is not in the document]
Comment (andrewdoro) [reply langchain/src/document_loaders/tests/unstructured.int.test.ts:30]: that's right, I've run the test locally using a docker instance of unstructured

File: langchain/src/document_loaders/fs/unstructured.ts [hunk 2]
```diff
@@ -175,19 +184,30 @@ export class UnstructuredLoader extends BaseDocumentLoader {
   private maxCharacters?: number;
 
   constructor(
-    filePathOrLegacyApiUrl: string,
+    filePathOrLegacyApiUrlOrMemoryBuffer:
```
Comment (jacoblee93) [inline langchain/src/document_loaders/fs/unstructured.ts:187]: We reeealy need to clean this up 😬

**Comment by jacoblee93** [comment]
Looks reasonable, thank you!

**Comment by jacoblee93** [comment]
Can't push to your branch - this lint rule is pretty silly, so you can either fix it or just add `"prefer-destructuring": 0,` to `langchain/eslintrc.cjs` and `libs/langchain-community/eslintrc.cjs`.

**Comment by andrewdoro** [comment]
@jacoblee93 fixed the formatting

**Comment by jacoblee93** [comment]
Thank you!

**Comment by jacoblee93** [comment]
Merged in https://github.com/langchain-ai/langchainjs/pull/5726

**Pull request closed without merging**